packages/db/src/test/networkGenealogy.spec.ts

```diff
@@ -0,0 +1,284 @@
+import {generateId, WorkspaceClient} from "./utils";
+import {
+  FindAncestors,
+  FindDescendants,
+  AddNetworkGenealogies
+} from "./networkGenealogy.graphql";
+import {AddNetworks} from "./network.graphql";
+
+// returnBlockHeader mocks the bare-bones return of a call to web3,
+// to check whether a network resource is a part of the network we're connected to
+const returnBlockHeader = blockNumber => {
+  const blockHeaders = {
+    1: {
+      number: 1,
+      hash: "0xcba0b90a5e65512202091c12a2e3b328f374715b9f1c8f32cb4600c726fe2aa6"
+    },
+    2: {
+      number: 2,
+      hash: "0xcba0b90a5e65512202091c12a2e3b328f374715b9f1c8f32cb4600c726fe2aa9"
+    },
+    3: {
+      number: 3,
+      hash: "0xcba0b90a5e65512202091c12a2e3b328f374715b9f1c8f32cb4600c726fe2aa1"
+    },
+    4: {
+      number: 4,
+      hash: "0xcba0b90a5e65512202091c12a2e3b328f374715b9f1c8f32cb4600c726fe2aa2"
+    },
+    5: {
+      number: 5,
+      hash: "0xcba0b90a5e65512202091c12a2e3b328f374715b9f1c8f32cb4600c726fe2aa3"
+    }
+  };
+
+  return blockHeaders[blockNumber];
+};
+
+describe("Network Genealogy", () => {
+  let wsClient;
+  let loneNetworkResult,
+    secondAncestorNetworkResult,
+    firstDescendantNetworkResult,
+    mainNetworkResourceResult;
+  let genealogyCheck;
+  let ancestors, descendants, addNetworkGenealogyRecord;
+
+  beforeAll(async () => {
+    wsClient = new WorkspaceClient();
+
+    // in reality this function would need some more looping through if there are more than
+    // 5 results but that logic can be handled in the loader tests
+    genealogyCheck = async (
+      id: string,
+      alreadyTried: [string],
+      type: string,
+      limit?: number
+    ) => {
+      let result;
+      let possibleMatches;
+
+      const queryLimit = limit ? limit : 5;
```

Comment:
JS has some nice syntax for default values you can use

```suggestion
      queryLimit: number = 5
    ) => {
      let result;
      let possibleMatches;
```